src/test/java/org/carlspring/cloud/storage/s3fs/fileSystemProvider/CheckAccessTest.java

```diff
@@ -105,59 +81,26 @@ void checkAccessWrite()
     }
 
     @Test
-    void checkAccessWriteDifferentUser()
-            throws IOException
+    void checkAccessMissingFile()
+        throws IOException
     {
         // fixtures
         final S3ClientMock client = S3MockFactory.getS3ClientMock();
         final String bucketName = "bucketA";
         final String directory = "dir";
-        final String filePath = directory + "/readOnly";
-        client.bucket(bucketName).dir(directory).file(filePath);
-
-        // return empty list
-        final Owner owner = Owner.builder().id("2").displayName("Read Only").build();
-        final GetObjectAclRequest request = GetObjectAclRequest.builder().bucket(bucketName).key(filePath).build();
-        final GetObjectAclResponse response = GetObjectAclResponse.builder().owner(owner).build();
-        doReturn(response).when(client).getObjectAcl(request);
-
-        final S3FileSystem fs = createNewS3FileSystem();
-        final String path = String.format("/%s/%s", bucketName, filePath);
-        final S3Path file1 = fs.getPath(path);
-
-        // We're expecting an exception here to be thrown
-        final Exception exception = assertThrows(AccessDeniedException.class,
-                                                 () -> s3fsProvider.checkAccess(file1, AccessMode.WRITE));
-
-        assertNotNull(exception);
-    }
-
-    @Test
-    void checkAccessWriteWithoutPermission()
-            throws IOException
-    {
-        // fixtures
-        final S3ClientMock client = S3MockFactory.getS3ClientMock();
-        final String bucketName = "bucketA";
-        final String directory = "dir";
-        client.bucket("bucketA").dir(directory);
-
-        // return empty list
-        final GetObjectAclResponse response = GetObjectAclResponse.builder().build();
-        final GetObjectAclRequest request = GetObjectAclRequest.builder().bucket(bucketName).key(
-                directory + "/").build();
-        doReturn(response).when(client).getObjectAcl(request);
+        client.bucket("bucketA");
 
         final String path = String.format("/%s/%s", bucketName, directory);
         final Path file1 = createNewS3FileSystem().getPath(path);
 
         // We're expecting an exception here to be thrown
-        final Exception exception = assertThrows(AccessDeniedException.class,
-                                                 () -> s3fsProvider.checkAccess(file1, AccessMode.WRITE));
+        final Exception exception = assertThrows(NoSuchFileException.class,
+            () -> s3fsProvider.checkAccess(file1, AccessMode.WRITE));
```

Comment:
Cosmetic: code style change.

```suggestion
        final Exception exception = assertThrows(NoSuchFileException.class,
                                                 () -> s3fsProvider.checkAccess(file1, AccessMode.WRITE));
```